example/node_example/example_with_full_possibility.js

```diff
@@ -0,0 +1,541 @@
+    const errorDeclaration = new ErrorDeclaration({
+        "declarationTime": "2020-01-15T00:00:00+01:00",
+        "reason": "incorrect_data",//todo
+        "example:vendorExtension": "Test1",
+        "correctiveEventIDs": ["urn:uuid:404d95fc-9457-4a51-bd6a-0bba133845a8"]
+    });
+
+    // readPoint
+    const readPoint = new ReadPoint({
+        "id": "urn:epc:id:sgln:0012345.11111.400"
+      });
+
+    // bizLocation
+    const bizLocation = new BizLocation();
+    bizLocation.setId("urn:epc:id:sgln:0012345.11111.0");
+
+    // bizTransactions
+    const bizTransaction = new BizTransactionElement();
+    bizTransaction.setType(cbv.businessTransactionTypes.po)
+    .setBizTransaction('urn:epc:id:gdti:0614141.00001.1618034');
+    const bizTransaction2 = new BizTransactionElement();
+    bizTransaction2.setType(cbv.businessTransactionTypes.pedigree)
+    .setBizTransaction('urn:epc:id:gsrn:0614141.00001.0000010253');
+    const bizTransaction3 = new BizTransactionElement();
+    bizTransaction3.setType(cbv.businessTransactionTypes.poc)
+    .setBizTransaction('urn:epc:id:gsrn:0614141.00001.0000023553');
+
+    const bizTransactionList = [ bizTransaction, bizTransaction2 ]
+
+    // quantity
+    const quantity = new QuantityElement();
+    quantity.setEpcClass("urn:epc:class:lgtin:4012345.012345.9911111")
+    .setQuantity(100)
+    .setUom('KGM');
+    const quantity2 = new QuantityElement();
+    quantity2.setEpcClass("urn:epc:class:lgtin:4012345.012345.992222")
+    .setQuantity(200)
+    .setUom('KGM');
+    const quantity3 = new QuantityElement();
+    quantity3.setEpcClass("urn:epc:class:lgtin:4012345.012345.993333")
+    .setQuantity(300)
+    .setUom('KGM');
+
+    const quantityList = [ quantity, quantity2 ]
+
+    // source
+    const source = new SourceElement();
+    source.setType(cbv.sourceDestinationTypes.location)
+    .setSource('urn:epc:id:sgln:4012345.00225.0');
+    const source2 = new SourceElement();
+    source2.setType(cbv.sourceDestinationTypes.possessing_party)
+    .setSource('urn:epc:id:sgln:4012345.00225.1');
+    const source3 = new SourceElement();
+    source3.setType(cbv.sourceDestinationTypes.owning_party)
+    .setSource('urn:epc:id:sgln:4012345.00225.2');
+
+    const sourceList = [ source, source2 ];
+
+    // destination
+    const destination = new DestinationElement();
+    destination.setType(cbv.sourceDestinationTypes.location)
+    .setDestination('urn:epc:id:sgln:0614141.00777.0');
+    const destination2 = new DestinationElement();
+    destination2.setType(cbv.sourceDestinationTypes.possessing_party)
+    .setDestination('urn:epc:id:sgln:0614141.00777.2');
+    const destination3 = new DestinationElement();
+    destination3.setType(cbv.sourceDestinationTypes.owning_party)
+    .setDestination('urn:epc:id:sgln:0614141.00777.3');
+
+    const destinationList = [ destination, destination2 ];
+
+    // sensor element
+        //sensor metadata
+        const sensorMetadata = new SensorMetadata();
```

Comment:
I would create functions for sub objects, such as buildSensorMetadataExample, buildSensorReportElementExample, ...